client/src/components/JurisdictionAdmin/BatchRoundSteps/DownloadStackLabelsButton.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { toast } from 'react-toastify'
+import * as Sentry from '@sentry/react'
+import { IButtonProps } from '@blueprintjs/core'
+import AsyncButton from '../../Atoms/AsyncButton'
+import useContestsJurisdictionAdmin from '../useContestsJurisdictionAdmin'
+import { downloadStackLabels } from '../generateSheets'
+import { sleep } from '../../../utils/sleep'
+
+interface DownloadStackLabelsButtonProps extends IButtonProps {
+  auditName: string
+  electionId: string
+  jurisdictionId: string
+  jurisdictionName: string
+}
+
+const DownloadStackLabelsButton = ({
+  auditName,
+  electionId,
+  jurisdictionId,
+  jurisdictionName,
+  ...buttonProps
+}: DownloadStackLabelsButtonProps): JSX.Element | null => {
+  const contestsQuery = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+
+  const onClick = async () => {
+    /* istanbul ignore next */
+    while (!contestsQuery.isSuccess) {
+      if (contestsQuery.isError) return
+      // eslint-disable-next-line no-await-in-loop
+      await sleep(100)
+    }
```

Comment:
The query implementation uses `fetch` which should have a default timeout, but what do you think about explicitly exiting this loop after a certain duration? The combination of while loop + sleep seems like it could leave us open to accidentally blocking for a long time.

Reply:
Hm yeah good point. I followed the pattern set in this [other button](https://github.com/votingworks/arlo/blob/main/client/src/components/JurisdictionAdmin/BatchRoundSteps/DownloadBatchTallySheetsButton.tsx#L34), but what you're saying does make sense and it does seem nice to be explicit about exiting after a certain duration. Though I don't have a good sense of what a more reasonable timeout might be, so I feel partially pulled to rely on the default `fetch` timeout and the existing pattern that we are relying on elsewhere we fetch the contests query to ensure behavior remains consistent

Reply:
It is a slightly scary looking pattern, I agree (I think I originally made it). However, the usual pattern would be to show some loading state instead of rendering the component until the data dependencies are loaded, which has equivalent risk of blocking for a long time. However, we don't usually worry about timeouts for something like that, since we're pretty confident the data will load just fine. So I'm not too worried here.